tests/invalid_models_tests/test_models.py

```diff
@@ -1313,8 +1356,8 @@ class Meta:
             obj=Model,
             id='models.W027',
         )
-        expected = [] if connection.features.supports_table_check_constraints else [warn]
-        self.assertCountEqual(errors, expected)
+        if not connection.features.supports_table_check_constraints:
+            self.assertIn(warn, errors)
```

Comment:
This change is not related with this patch, please revert it.

Reply:
Hm, I think I suggested this change on the new test, did not realize the existing test used `assertCountEqual`. Still, this looks like a good improvement, but then in a separate commit or PR (the commit history of this PR should probably be squashed anyway before merging).

Reply:
It's not an improvement. We run tests on databases that support check constraints and on databases that don't support them. We want to check both cases.

Reply:
Right, I think the improvement part is that you're actually looking at the content of the errors, rather than just at the count. With the original code, there really is no point in recreating the full warning literally, you could also just say:

```
expected = 0 if connection.features.supports_table_check_constraints else 1
self.assertEqual(len(errors), expected)
```

Actually checking the content of the error sounds like an improvement to me. But the proposed change actually checks *nothing* when the database does not support check constraints, which is indeed not good.

So, how about just replacing the `assertCountEqual` with an `assertEqual`, e.g.:

```
expected = [] if connection.features.supports_table_check_constraints else [warn]
        self.assertEqual(errors, expected)
```

Using `assertIn` rather than `assertEqual` also seemed reasonable (since there might be more warnings (maybe not now, then later). But then you would need to check either `assertIn` or `assertNotIn`, e.g.:

```
if not connection.features.supports_table_check_constraints:
            self.assertIn(warn, errors)
else:
            self.assertNotIn(warn, errors)
```

This seems to be the most complete and generic check?

Reply:
> Right, I think the improvement part is that you're actually looking at the content of the errors, rather than just at the count. With the original code, there really is no point in recreating the full warning literally, you could also just say:
> 
> ```
> expected = 0 if connection.features.supports_table_check_constraints else 1
> self.assertEqual(len(errors), expected)
> ```

[`assertCountEqual()`](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertCountEqual) doesn't work that way, but do to the fact that we have list with only 1 element we can use `self.assertEqual(errors, expected)`.

Reply:
So, is `assertCountEqual()` like counting how many elements inside a list that equals to the object we're checking, not the whole length of list?
If that's so, I think using `assertCountEqual()` is a good option.

Reply:
Oh, confusing name then, `assertCountEqual()`, thanks for the link.

So then this is really not much of an improvement, unless the original code fails because there is more than one one error (but given only a very specific model is checked, it should not normally raise additional warnings, and if it does, failing the test to force review of the testcase is probably a good idea).

So indeed, reverting this change would be best. Sorry for the noise :-)